### PR TITLE
Implemented cvar[] - access exported C variables.

### PR DESCRIPTION
**Pull request by abalkin**
Added a .ffi.cv function that takes the same type/name pair as .ffi.cf, but returns the named symbol as a value.  See [Accessing values exported from dlls](https://docs.python.org/3/library/ctypes.html#accessing-values-exported-from-dlls) for use cases.

### CODE REVIEW

**Comment by sv** [comment]
Do you have any usage examples for this?

**Comment by abalkin** [comment]
Yes, grabbing the value of Py_OptimizeFlag from libpython is the canonical example from ctypes that I linked in my first comment.  Another example is the GNU readline library that exports many of its [variables](https://cnswww.cns.cwru.edu/php/chet/readline/readline.html#SEC28).

File: ffi.c [hunk 1]
```diff
@@ -435,7 +435,21 @@ EXP K deref(K x) {
   return r;
 }
 
-#define N 9
+Z K1(cv){
```
Comment (sv) [inline ffi.c]: rename to`cvar`
define as `EXP K cvar(K x)` so it has a chance on windows

File: ffi.c [hunk 2]
```diff
@@ -435,7 +435,21 @@ EXP K deref(K x) {
...
+  V *v;
+  H t;
+  if (xt==-KS) {
+    t = KI;
```
Comment (sv) [inline ffi.c]: `t=-KI`

File: ffi.c [hunk 3]
```diff
@@ -435,7 +435,21 @@ EXP K deref(K x) {
   return r;
 }
 
-#define N 9
+Z K1(cv){
+  V *v;
+  H t;
+  if (xt==-KS) {
```
Comment (sv) [inline ffi.c]: use direct access `x->t`

File: ffi.c [hunk 4]
```diff
@@ -435,7 +435,21 @@ EXP K deref(K x) {
...
+    t = KI;
+  } else {
+    P(xt||xn!=2||xK[0]->t!=-KC,krr("type"));
```
Comment (sv) [inline ffi.c]: use simple if and direct access to members. `xK->kK(x)` to match rest of the file

**Review by sv — CHANGES_REQUESTED** [review]

**Comment by abalkin** [comment]
Sergey,

I addressed your comments.

File: ffi.c (hunk 1 above)
Comment (abalkin) [reply ffi.c]: Are you sure it is necessary to export functions that are exposed via `dl`? I don't know much about Windows, but Python uses a similar mechanism to export functions from C modules and they use addresses of static functions to populate method lookup tables.

**Pull request merged**